miden/src/cli/run.rs

```diff
@@ -100,6 +104,18 @@ impl RunCmd {
 // HELPER FUNCTIONS
 // ================================================================================================
 
+pub fn create_execution_options(cmd: &RunCmd) -> Result<ExecutionOptions, Report> {
```

Comment:
I think this function does not need to be public.

But also, related to the last comment, it may be simpler to move this back into the `run_program()` function and do something like:
```rust
let mut execution_options =
    ExecutionOptions::new(Some(cmd.max_cycles), cmd.expected_cycles, cmd.tracing)
        .into_diagnostic()?;

let program = if cmd.debug {
    execution_options = execution_options.with_debugging();
    ProgramFile::read(&params.assembly_file)?.compile(&Debug::On, &libraries.libraries)?
} else {
    ProgramFile::read(&params.assembly_file)?.compile(&Debug::Off, &libraries.libraries)?
}
```

Reply:
It would be a little cleaner to add `enable_debugging` to the [constructor arguments of ExecutionOptions](https://github.com/0xPolygonMiden/miden-vm/blob/fbce798a53967310463ee02e0e681617781406df/air/src/options.rs#L191)

Then we can easily avoid a conditional making the code more straightforward to read:
```python
let execution_options = ExecutionOptions::new(Some(params.max_cycles), params.expected_cycles, params.tracing, 
params.debug).into_diagnostic()?;
```

Reply:
> It would be a little cleaner to add `enable_debugging` to the [constructor arguments of ExecutionOptions](https://github.com/0xPolygonMiden/miden-vm/blob/fbce798a53967310463ee02e0e681617781406df/air/src/options.rs#L191)

Yes, agreed. At the time when we introduced `enable_debugging` we needed to do it without making it a breaking change, but I think it is fine to fix this now (the next release will contain a number of breaking changes anyway).

So - let's update the constructor to include `enable_debugging`.

> For the `Debug` enum we can implement the `From` trait, and do something like this:
> 
> ```python
> let program =
>         ProgramFile::read(&params.assembly_file)?.compile(&Debug::from(params.debug), &libraries.libraries)?;
> ```
> 
> again without a need for a conditional making this high level function easer to follow.

Yes, that would work well too. We may even be able to do it like:
```rust
let program =
    ProgramFile::read(&params.assembly_file)?.compile(params.debug.into(), &libraries.libraries)?;
```